projectgenerator/libqgsprojectgen/dbconnector/gpkg_connector.py

```diff
@@ -26,11 +26,46 @@
 class GPKGConnector(DBConnector):
     def __init__(self, uri, schema):
         DBConnector.__init__(self, uri, schema)
-        self.conn = qgis.utils.spatialite_connect(uri)
+        self.conn = self._get_connection(uri)
         self.conn.row_factory = sqlite3.Row
         self._bMetadataTable = self._metadata_exists()
         self._tables_info = self._get_tables_info()
 
+    def _get_connection(self, uri):
+        '''
+        Get a dbapi2.Connection to GeoPackage.
+
+        Heavily based on QGIS qgis.utils.spatialite_connect, but reimplemented
+        to avoid a Windows crash using pyspatialite with GeoPackage.
```

Comment:
Is that something that should be fixed upstream?

Reply:
Definitely, I'll create a PR for that :)

Reply:
:+1: in this case I think we can kill this patch here again to avoid duplicated code?

Reply:
Sure! When such PR gets merged :)

Reply:
You know how to bargain, that's for sure ;)
bring it on

Reply:
Jaja, nah, not part of my abilities.
The crash is already reported https://issues.qgis.org/issues/16720
I think the solution deserves some discussion first in the mailing-list before a QGIS PR.

Reply:
is there something about this code that needs to be discussed first? what?

Reply:
After reading the discussion: Would be great if you could open a pull request. Also [reading this](https://groups.google.com/d/msg/spatialite-users/o0jUwMUqx_g/pfZBDTrWAAAJ)
 
> (...)
> Almost all distros now support module loading, making pyspatialite completely unnecessary.  If one doesn’t, that should be fixed at this point (...)

Reply:
Well,  cur.execute("select load_extension('{}', '{}')".format(lib, entry_point)) is failing for me on Windows for all options given [here](https://github.com/qgis/QGIS/blob/master/python/utils.py#L602-L609). That's why I wanted to know QGIS devs opinion first. 

So, you think I can go ahead and remove [python/ext-libs](https://github.com/qgis/QGIS/tree/master/python/ext-libs) from QGIS core? I can do that in the PR, along with the adjusted qgis.utils.spatialite_connect().

Reply:
Just try, it will still depend on all platforms (that probably means: Windows) shipping sqlite with the possibility to load external modules. But having this ready will likely improve the chances that this happens.

Reply:
Thanks for the pull request, now that this got accepted I assume it can be removed from this pull request?

Reply:
Yesterday I tested a Generate operation from ili to GeoPackage with [OSGeo4W release # 44](http://qgis.org/downloads/weekly/QGIS-OSGeo4W-2.99.0-44-Setup-x86_64.exe) and even though the QGIS crash disappeared, the SpatiaLite module couldn't be loaded to `sqlite3`, so there's a problem that still remains and makes [d1c614a](https://github.com/opengisch/projectgenerator/pull/112/commits/d1c614a4dcf4bafe2b1a280d613a57d783509bf3) a must. I'm checking how the upstream fix evolves to revert this commit.

Reply:
Might be an additional install inside osgeo4w?
@jef-n is there anything to do manually for mod_spatialite?

Reply:
Just reverted the commit as it is now safe to use `qgis.utils.spatialite_connect()` on Windows as well.